src/frontends/tensorflow/src/frontend.cpp

```diff
@@ -474,6 +479,12 @@ std::shared_ptr<ov::Model> FrontEnd::decode(const ov::frontend::InputModel::Ptr&
 }
 
 void FrontEnd::convert(const std::shared_ptr<ov::Model>& partiallyConverted) const {
+    std::map<std::string, std::unordered_set<std::string>> names;
+
+    for (const auto& result : partiallyConverted->get_results()) {
+        names[result->get_friendly_name()] = result->get_output_tensor(0).get_names();
+    }
```

Comment:
why do we need this? Looks like wa. Below is the same comment

Reply:
This is needed to save output names from ov.Model() before decoding and restore them after work of translator's.  Otherwise translator's rewrite them in 490 line. I think it should be fixed if we align inner names in set_node_name().

Reply:
yes, let us do it `set_node_name`

Reply:
As discussed offline if we fix set_node_name, NMS stops to work, where wrong tensor names are set for some reason. I suggest to fix it in separate PR.

Regurding WA in convert(), I removed it as agreed and corrected test that compares convert and decode+convert paths: https://github.com/openvinotoolkit/openvino/pull/21075/files#diff-16b422a3f3ea0023255bea536b7fb6ae89da562ebc7057c871f9fba35711761b

Reply:
The problem with NMS problem was because outputs of NMS have "name:1", "name:2"... indices in op names. And `set_node_name()` added additional :0 which resulted in "name:1:0", "name:2:0". The problem was fixed when I corrected `set_node_name()` to not add :0 for names with indices.

But there is another problem with `set_node_name()`. This method sets names for model `Parameter`'s. This leads to problem in MO preprocessing. MO preprocessing tries to find model input names among `Place`'s, which are set by operation names: 
https://github.com/openvinotoolkit/openvino/blob/c6258294fea969d2908efe841df068bdb98d2b88/src/frontends/tensorflow/src/input_model.cpp#L224C14-L224C14

So looks like we need to create `Place`'s in `InputModel` with tensor names to make preprocesssing work.